Unix/doc/diagnose-provider-bugs.md

```diff
@@ -0,0 +1,184 @@
+# Diagnosing OMI Provider Failures
+
+At times, OMI providers may fail. Since multiple OMI providers can be
+serviced using a single `omiagent` process, it can be difficult to
+diagnose the exact provider that is causing the source of the problem.
+
+This documentation is intended to help diagnose what provider is
+triggering failures, allowing support staff to quickly route problems
+to the appropriate team for support/resolution.
+
+### Minimum OMI Version
+
+**NOTE: This document assumes OMI v1.4.1-0 or later.** If you have a
+prior version of OMI, we suggest you update to the latest version of
+OMI available on the
+[OMI releases](https://github.com/Microsoft/omi/releases)
+page. The latest version of OMI should be able to be installed on top
+of your existing OMI version with no significant negative consequences.
+
+### Caviats
+
+OMI v1.4.1-0 has some issues logged against it regarding logging:
+
+1. Issue #322 - "Leaky" OMI providers can be difficult to isolate
+2. Issue #449 - OMI logs ERROR message (NTLM creds) in normal case
+3. Issue #450 - OMI logs warnings during enumeration in non-error cases
+4. Issue #451 - OMI server configuration file needs to be cleaned up
+5. Issue #452 - OMI Logging: OMI log file is MUCH too chatty in INFO mode
+
+Issue #322 means that this document is primarily written to isolate
+providers that are crashing or abnormally terminating. Providers that
+are not crashing, but have other ill effects (excessive memory growth,
+for example), are still difficult to isolate.
+
+The remaining issues noted above means that OMI log files will grow
+more than usual, perhaps requiring log file rotation of OMI's log
+files. This will be resolved in a future release.
+
+Finally, be aware that `EventId` identifiers in OMI tend to change
+from release to release. Thus, they are not a good way to isolate
+specific log file entries. As a result, this document will use `grep`
+commands searching for text of log file entries rather than specific
+`EventId` identifiers.
+
+### Enabling logging in OMI
```

Comment:
Would not change in the conf file require omiserver restart to take effect?

Reply:
we need restart omiserver to take effect.